components/governance/org.wso2.carbon.governance.taxonomy/src/main/java/org/wso2/carbon/governance/taxonomy/util/CommonUtils.java

```diff
@@ -0,0 +1,266 @@
+
+        if (nodeList.getLength() == 0) {
+            throw new TaxonomyException("No results for provided query : " + query + ".Please provide valid query.");
+        }
+
+        itemRelativeRoot.put(ELEMENT_ID, ((Element) nodeList.item(0).getParentNode()).getAttribute(ID));
+        itemRelativeRoot.put(TEXT, ((Element) nodeList.item(0).getParentNode()).getAttribute(DISPLAY_NAME));
+
+        int nodeCount;
+        if (nodeList.getLength() > endNode) {
+            nodeCount = endNode;
+        } else {
+            nodeCount = nodeList.getLength();
+        }
+
+        JSONObject dataArray;
+        if (!updatedQuery.equals("/taxonomy/root/*")) {
+            // this will execute when there is long path
+            for (int i = startNode; i < nodeCount; ++i) {
+                dataArray = new JSONObject();
+
+                String currentId = ((Element) nodeList.item(i)).getAttribute(ID);
+
+                dataArray.put(ELEMENT_ID, currentId);
+                //                dataArray.put(ID, currentId);
+                dataArray.put(TEXT, ((Element) nodeList.item(i)).getAttribute(DISPLAY_NAME));
+                dataArray.put(CHILDREN, nodeList.item(i).hasChildNodes());
+
+                if (nodeCount == 1) {
+                    dataArray.put(PREVIOUS_ELEMENT_PATH, "");
+                    dataArray.put(CURRENT_ELEMENT_PATH, query.substring(0, query.lastIndexOf("/")) + "/" + currentId);
+                    dataArray.put(NEXT_ELEMENT_PATH, "");
+                } else if (i == 0) {
+                    dataArray.put(PREVIOUS_ELEMENT_PATH, "");
+                    dataArray.put(CURRENT_ELEMENT_PATH, query.substring(0, query.lastIndexOf("/")) + "/" + currentId);
+                    dataArray.put(NEXT_ELEMENT_PATH,
+                            query.substring(0, query.lastIndexOf("/")) + "/" + ((Element) nodeList.item(i + 1))
+                                    .getAttribute(ID));
+                } else if (i == nodeCount - 1) {
+                    dataArray.put(PREVIOUS_ELEMENT_PATH,
+                            query.substring(0, query.lastIndexOf("/")) + "/" + ((Element) nodeList.item(i - 1))
+                                    .getAttribute(ID));
+                    dataArray.put(CURRENT_ELEMENT_PATH, query.substring(0, query.lastIndexOf("/")) + "/" + currentId);
+                    dataArray.put(NEXT_ELEMENT_PATH, "");
+                } else {
+                    dataArray.put(PREVIOUS_ELEMENT_PATH,
+                            query.substring(0, query.lastIndexOf("/")) + "/" + ((Element) nodeList.item(i - 1))
+                                    .getAttribute(ID));
+                    dataArray.put(CURRENT_ELEMENT_PATH, query.substring(0, query.lastIndexOf("/")) + "/" + currentId);
+                    dataArray.put(NEXT_ELEMENT_PATH,
+                            query.substring(0, query.lastIndexOf("/")) + "/" + ((Element) nodeList.item(i + 1))
+                                    .getAttribute(ID));
+
```

Comment:
Thank you Thushara for mentioning that, 
I have fixed it with this PR https://github.com/wso2/carbon-governance/pull/253